### ISSUE TITLE
Improve messaging for "image-alt"(has-alt) failures on elements other than <area>, <input> or <img>

### ISSUE THREAD

**Issue by iamrafan** [dequelabs/axe-core#1192]
![image](https://user-images.githubusercontent.com/22068434/47054651-e47c7b00-d167-11e8-9abe-46c8b73721d8.png)

- Link to live demo: https://codepen.io/iamrafan/pen/JmpPRR?editors=1000#0

<pre><code>axe-core version: 3.1.1
axe-Coconut version: 3.5.0</code></pre>

**Comment by WilcoFiers** [comment]
Taking off the "false positive" thing. Axe correctly identifies this `span` as being inaccessible. What it does incorrectly here is recommend to add "alt". We've seen a similar problem with `svg[role=img]`, where someone added an alt to that and axe kept flagging that. Axe does the right thing, but its remediation suggestion is wrong.

What we should do is add a new rule that looks specifically at `role=img`, so we can report the remediation properly.

**Comment by marcysutton** [comment]
# Role-Img-Alt

Inspect elements with an explicit `role="img"` for missing compatible text alternatives. This rule is intended to provide better messaging than the existing image-alt rule, which only applies to elements with support for the `alt` attribute.

Help text: Element with role=img must provide a text alternative

**Tags:** cat.text-alternatives, wcag2a, wcag111, section508, section508.22.a

## Selector

`role="img"`

## Checks

`any:  "aria-label", "aria-labelledby", "non-empty-title"`

**Comment by AutoSponge** [comment]
Looking at the rules we currently have, I think the best path forward is to remove `[role='img']:not(svg)` from the `image-alt` rule and create a separate rule for `svg-alt` and `role-img-alt`. This would follow how `area-alt`, `input-image-alt`, and `object-alt` are configured.

While `role="img"` is redundant on an `<img>`, the advice to add `alt` still holds. So, no changes to `img-alt` other than removing `[role="img"]`.

`svg-alt` would use an `svg[role="img"]` selector to avoid violations/warns for `role=none` and `role=presenation`. I feel like this should be a separate rule because a `title` is sufficient (see https://www.deque.com/blog/creating-accessible-svgs/). Other techniques can be referenced or listed. I feel this requires some discussion since the link referenced does not include cases of `<svg><use>...</svg>`

`role-img-alt` would use `[role="img"]:not(svg):not(img):not(area):not(input):not(object)` selector. This would limit techniques to `aria-label` and `aria-labelledby`.

**Comment by WilcoFiers** [comment]
We don't need to do the svg-alt to fix this. That's part of other work. But I agree with the role-img-alt idea, although I think Marcy is right that that does still need the non-empty-title check. All HTML elements can get an accessible name from the title attribute.

**Comment by AutoSponge** [comment]
> All HTML elements can get an accessible name from the title attribute.

That's not consistent with rules like "Buttons must have discernible text" which fails this markup:

```html
<button title="foo"><svg><use xlink:href="#any"></use></svg></button>
```

By your statement, this should pass but it doesn't:

```html
<button><svg title="foo"><use xlink:href="#any"></use></svg></button>
```

While `title` works in the accName calc, I'm not sure non-empty `title` is considered as widely as "all HTML elements" within axe-core.

**Comment by jeankaplansky** [comment]
New rule help required for axe-core 3.3

**Comment by jeankaplansky** [comment]
role-img-alt rule created, and rule help is here: https://dequeuniversity.com/rules/axe/3.3/role-img-alt. We do not go into details beyond what we've put in the correct and incorrect markup examples.

**Comment by somaalapati** [comment]
@chandana7393 Let me know if you need help to verify this ticket.

**Comment by chandana7393** [comment]
Tested, working as expected.
![role_img_alt](https://user-images.githubusercontent.com/45285854/62458065-a7868100-b799-11e9-9e61-5315f8a5dd68.PNG)
![img_alt](https://user-images.githubusercontent.com/45285854/62458069-a9504480-b799-11e9-988d-12123ebc7b28.PNG)
![area_alt](https://user-images.githubusercontent.com/45285854/62458071-abb29e80-b799-11e9-9da1-bf54396a28a6.PNG)
![input_alt](https://user-images.githubusercontent.com/45285854/62458074-ad7c6200-b799-11e9-9d28-8f63d61d8707.PNG)
**Tested Environment:**
Attest - 2.5.1.21115v
Axe - 3.8.1.21115v
Axe-coconut - 3.8.1.21304v
Chrome - 76.0.3809.87v
Firefox - 68.0.1v
OS - Windows 10 64 bit.
